lib/apiservers/engine/backends/container.go

```diff
@@ -421,49 +421,46 @@ func (c *ContainerBackend) ContainerExecStart(ctx context.Context, eid string, s
 		actor := CreateContainerEventActorWithAttributes(vc, map[string]string{})
 		EventService().Log(event, eventtypes.ContainerEventType, actor)
 
-		// no need to attach for detached case
-		if !attach {
-			op.Debugf("Detached mode. Returning early.")
-			return nil
-		}
+		if attach {
```

Comment:
This does not appear to change logic flow - should be reverted unless there's a solid reason because:
a. idiom is for early return where possible, avoiding nesting code blocks/indenting where possible
b. confuses the PR.

The only possibility I see is that you're changing it so that even if `attach == false` we will wait for exit. I do not think that's the semantic expected for `exec -d`, but please confirm.

Reply:
Will confirm, that was my original assumption. But with the mention of `exec -d` I am now skeptical. I will just revert for now most likely.